spec/Staking.md

```diff
@@ -0,0 +1,368 @@
+
+### Example
+
+`REVOKE_PENDING_DURATION` is 500 in this example.
+
+```
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0xAB..CD": 690,
+  },
+  delegation: {
+    "0x23..45": [ ["0x01..23", 500] ],
+    "0xAB..CD": [ ["0x01..23", 200], ["0x23..45", 110] ]
+  }
+}
+
+> "0xAB..CD" sends staking action [3, "0x01..23", 100] at block timestamp 123400
+
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0xAB..CD": 690,
+  },
+  delegation: {
+    "0x23..45": [ ["0x01..23", 500] ],
+    "0xAB..CD": [ ["0x01..23", 200], ["0x23..45", 110] ]
+  },
+  revocations: [ ["0xAB..CD", "0x01..23", 123900, 100] ]
+}
+
+> "0xAB..CD" sends staking action [3, "0x01..23", 50"] at block timestamp 123500
+
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0xAB..CD": 690,
+  },
+  delegation: {
+    "0x23..45": [ ["0x01..23", 500] ],
+    "0xAB..CD": [ ["0x01..23", 200], ["0x23..45", 110] ]
+  },
+  revocations: [
+    ["0xAB..CD", "0x01..23", 123900, 100],
+    ["0xAB..CD", "0x01..23", 124000, 50],
+  ]
+}
+
+> "0x23..45" sends staking action [3, "0x01..23", 500"] at block timestamp 123600
+
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0xAB..CD": 690,
+  },
+  delegation: {
+    "0x23..45": [ ["0x01..23", 500] ],
+    "0xAB..CD": [ ["0x01..23", 200], ["0x23..45", 110] ]
+  },
+  revocations: [
+    ["0xAB..CD", "0x01..23", 123900, 100],
+    ["0xAB..CD", "0x01..23", 124000, 50],
+    ["0x23..45", "0x01..23", 124100, 500],
+  ]
+}
+
+> after a block with timestamp 123900
+
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0xAB..CD": 790,
+  },
+  delegation: {
+    "0x23..45": [ ["0x01..23", 500] ],
+    "0xAB..CD": [ ["0x01..23", 100], ["0x23..45", 110] ]
+  },
+  revocations: [
+    ["0xAB..CD", "0x01..23", 124000, 50],
+    ["0x23..45", "0x01..23", 124100, 500],
+  ]
+}
+
+> after a block with timestamp 124000
+
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0xAB..CD": 840,
+  },
+  delegation: {
+    "0x23..45": [ ["0x01..23", 500] ],
+    "0xAB..CD": [ ["0x01..23", 50], ["0x23..45", 110] ]
+  },
+  revocations: [
+    ["0x23..45", "0x01..23", 124100, 500],
+  ]
+}
+
+> after a block with timestamp 124100
+
+state = {
+  stakeholders: [ "0x23..45", "0xAB..CD" ],
+  balance: {
+    "0x23..45": 500,
+    "0xAB..CD": 840,
+  },
+  delegation: {
+    "0xAB..CD": [ ["0x01..23", 50], ["0x23..45", 110] ]
+  }
+}
+```
+
+# Fee distribution
+
+You pay fees to make a transaction, which is greater than the specified minimum fee for the transaction type.
+Transaction fees that are within a block are collected, and CCS holders and the block author share this.
+
+CCS Holders share the sum of the minimum fees for the transaction.
+They get fees in proportion to the CCS balance they have which is the sum of the amount of undelegated CCS plus the total amount of CCS that are delegated to someone.
```

Comment:
They get fees in proportion to the CCS balance they have, which is the sum of the amount of undelegated CCS plus the total amount of CCS that are delegated to someone.